### DISCUSSION TITLE
Why pipe gets a value as first arg and why we have now flow?

### DISCUSSION THREAD

**Discussion by deleite** [gcanti/fp-ts#1753] [Q&A]
I learned the basics of fp from Dr boolean mostly adequate. There compose is defined and pipe can be derived from it. But in every place it is just a flipped compose.

What is the difference here with pipe and flow?

**Comment by steida** [comment]
https://rlee.dev/practical-guide-to-fp-ts-part-1

I recommend starting with pipe, and sooner or later, you will see that some pipes can be replaced with the flow for the more concise code.

**Reply by deleite** [reply]
Thanks, but that does not answer my question. In other fp libraries pipe is not defined like this and flow is actually pipe. So what is the motivation here? And where the naming comes from? Are there performance benefits for this?

**Reply by gcanti** [reply]
> where the naming comes from?

- "pipe": from the pipe operator (`|>`)
- "flow": from lodash (https://lodash.com/docs/4.17.15#flow)
